### ISSUE TITLE
failed to create "consolelinks" warning logs on jaeger operator on OpenShift

### ISSUE THREAD

**Issue by jkandasa** [jaegertracing/jaeger-operator#1308]
#### Steps to reproduce:
* Install community jaeger operator
* deploy jaeger services on a namespace
* watch jaeger-operator log

#### Openshift version 
```
$ oc version
Client Version: openshift-clients-4.6.0-202006250705.p0-75-g39d7f091b
Server Version: 4.6.3
Kubernetes Version: v1.19.0+9f84db3
```

#### operator version:
```
$ oc get csv
NAME                                           DISPLAY                             VERSION                 REPLACES                                       PHASE
jaeger-operator.v1.20.0                        Community Jaeger Operator           1.20.0                  jaeger-operator.v1.19.0                        Succeeded
```

#### Log:
```
time="2020-11-19T06:16:54Z" level=warning msg="failed to reconcile console links" error="consolelinks.console.openshift.io is forbidden: User \"system:serviceaccount:openshift-operators:jaeger-operator\" cannot create resource \"consolelinks\" in API group \"console.openshift.io\" at the cluster scope" instance=jaegerqe namespace=jk
time="2020-11-19T06:16:55Z" level=warning msg="failed to reconcile console links" error="consolelinks.console.openshift.io is forbidden: User \"system:serviceaccount:openshift-operators:jaeger-operator\" cannot create resource \"consolelinks\" in API group \"console.openshift.io\" at the cluster scope" instance=jaegerqe namespace=jk
```

**Comment by rubenvp8510** [comment]
Could you confirm what kind of permissions do you have?

IIRC consolelinks are cluster resources.

**Comment by jpkrohling** [comment]
Even if the operator is installed with namespace-only permissions, the operator shouldn't fail the reconciliation because of the console links. The operator should:

1. determine whether it should manage console links
1. not break the reconciliation console links fail to reconcile
1. perhaps add a message to the CR with the error? Do we have something like this already for this operator?

**Comment by rubenvp8510** [comment]
I tried to reproduce this issue installing an operator that watch for only one name space, I wasn't able to reproduce the issue in that way. Then I tried installing a cluster-wide operator and I was able to reproduce it.

I think the issue is because the permissions weren't added to the generated CSV.  PR that fixes the issue: https://github.com/jaegertracing/jaeger-operator/pull/1177

BTW, the reconciliation is not broken (or shouldn't be) it's just a warning message. 
https://github.com/jaegertracing/jaeger-operator/blob/b7d3cfd5ef458702c3a25c797dcee828fd75cc6a/pkg/controller/jaeger/jaeger_controller.go#L343
